doc/running-cljdoc-locally.adoc

```diff
@@ -0,0 +1,461 @@
+
+=== Prerequisites
+You'll need https://www.docker.com/get-started[docker].
+
+=== Setup cljdoc docker
+TIP: Your installation might require you to prefix docker commands with `sudo`, add your user to the docker group or adapt as necessary.
+
+1. Fetch any updates for the docker image so that we match what is running in production:
++
+[source,shell,subs="verbatim,attributes"]
+----
+docker pull {docker-image}
+----
+
+2. Make a directory for the cljdoc sqlite database:
++
+[source,shell]
+----
+mkdir -p /tmp/cljdoc
+----
+
+3. Start the cljdoc docker web server:
++
+[source,shell,subs="verbatim,attributes"]
+----
+docker run --rm \
+  --publish 8000:8000 \
+  --volume "$HOME/.m2:/root/.m2" \
+  --volume /tmp/cljdoc:/app/data \
+  {docker-image}
+----
++
+NOTE: Leave this terminal open. Open a separate terminal shell for any other work.
+
+[[import-a-library,import a library's docs]]
+== Import a Library's Docs
+Before importing a library for local testing, you will have decided if you want to <<run-cljdoc-locally-from-source>> or <<run-cljdoc-locally-from-docker>>.
+See the <<introduction>> section on how to make that choice.
+
+Here's an overview of importing library docs when running cljdoc locally:
+
+image::running-local.svg[running locally overview]
+
+Note:
+
+* A *cljdoc developer* won't necessarily want to work with a local library; they can skip to step 3 and <<import-a-published-library>>.
+* In the discussions that follow, we'll assume that your hosted SCM (Source Code Management) system is GitHub.
+
+=== Example Library
+We have chosen {example-project-link}, {example-project-desc}, as our example import library.
+Feel free to substitute your own, or another, library when walking through examples.
+
+=== The ingest Command
+The cljdoc command line tool includes an `ingest` command.
+You will use this command to import a library's documentation into cljdoc when working locally.
+
+All example commands assume you are running from your library project root dir and that, if you are running from cljdoc source, both projects share the same parent dir.
+
+The sections that follow include example command variants for running from cljdoc source and docker.
+For example to get help on the ingest command:
+
+.cmd for cljdoc source
+[%collapsible]
+====
+=====
+[source,shell,subs="verbatim,attributes"]
+----
+../cljdoc/script/cljdoc ingest --help
+----
+=====
+====
+
+.cmd for cljdoc docker
+[%collapsible]
+====
+=====
+[source,shell,subs="verbatim,attributes"]
+----
+docker run --rm --entrypoint clojure {docker-image} -A:cli ingest --help
+----
+=====
+====
+
+TIP: Your installation might require you to prefix docker commands with `sudo`, adapt as necessary.
+
+=== Steps
+
+Repeat steps 1 through 4 as needed (see overview diagram).
+Confusion can usually be resolved by asking:
+
+* Did I remember to commit/push?
+* Did I remember to publish locally?
```

Comment:
I think this might be a good candidate for an admonition `TIP` block now that this is an asciidoc file :) https://asciidoctor.org/docs/asciidoc-syntax-quick-reference/#admon-bl

Reply:
Sure, makes sense, will do!